src/gc/env/gcenv.ee.h

```diff
@@ -68,6 +68,12 @@ class GCToEEInterface
     static void StompWriteBarrier(WriteBarrierParameters* args);
 
     static void EnableFinalization(bool foundFinalizers);
+
+    // Global alloc context retrieval
+    static gc_alloc_context *GetGlobalAllocContext();
```

Comment:
If you do that, using_allocation_contexts can become the pointer to the global allocation context instead of a flag - it will make the updates a tiny bit cheaper.